CONTRIBUTING.md

```diff
@@ -1,65 +1,189 @@
+		$ cd build
+		$ cmake --build . --config Release
+4. Set (add) the remote repository:
+
+        $ git remote add exiv2_upstream https://github.com/Exiv2/exiv2.git
+
+	and verify that you have the two remotes configured correctly:
+
+        $ git remote -v
+        exiv2_upstream  https://github.com/Exiv2/exiv2.git (fetch)
+        exiv2_upstream  https://github.com/Exiv2/exiv2.git (push)
+        origin  https://github.com/your-username/exiv2.git (fetch)
+        origin  https://github.com/your-username/exiv2.git (push)
+		
+5. Next, create a branch for your PR:
+
+        $ git checkout -b my_first_pr_branch
+
+6. Make your change(s), add tests for your changes, and commit each change:
+
+        ...
+        
+        $ git commit -m "Commit message 1"
+        
+        ...
+        
+        $ git commit -m "Commit message 2"
+
+7. Make sure the tests pass:
+
+        $ make tests         # Application tests
+        $./bin/unit_tests    # Unit tests
+
+	Exiv2's test system is described on more detail in Appendix [A.1](#a1-the-exiv2-test-system) (TODO)
+
+8. Push the changes to your fork on GitHub:
+
+        $ git push origin my_first_pr_branch
+
+9. Create the PR by pressing the _New pull request_ button on: `https://github.com/your-username/exiv2`
+
+
+10. Now wait for an exiv2 project member(s) to respond to your PR. Follow the discussion on your PR at [https://github.com/Exiv2/exiv2/pulls](GitHub). 
+   You may have to do some updates to your PR until it gets accepted.
 
-4. Make your change. Add tests for your change. Make the tests pass.
-5. Create the Pull Request (PR).
-    a. Check if the CI checks pass.
-    b. If your PR lives for a long time, try to avoid to press the button "Update branch" in the Pull Request view. We prefer rebasing in top of master, instead of merging master into branches.
-    c. A project member will review your PR as soon as possible.
 
-## Pull Request guidelines
-- PRs should be kept at a manageable size. Try to focus in just one goal per PR. If you find yourself doing several things that in a PR that were not expected, try to split the different tasks in different PRs.
-- Commits should always change one logical unit, so that cherry-picking & reverting is simple.
-- Commit messages should as informative and concise as possible. The first line of the commit message should be < 80 characters and describe the commit briefly. If the 80 characters are too short for a summary, consider splitting the commit. Below the short summary add one blank line and then a more detailed explanation if required.
+11. After the PR has been reviewed you must _rebase_ your repo copy since there may have been several changes to the main upstream repo 
+   (a PR can live for several days or even weeks).
 
-## Code guidelines ##
-### General ###
+	First, fetch stuff from exiv2_upstream (it does not matter in which branch you are currently in on your local copy of the repository):
+
+        $ git fetch exiv2_upstream
```

Comment:
This is a quite verbose equivalent of `git checkout my_first_pr_branch && git pull --rebase origin master && git push --force`. I'd rather keep it short as there are fewer things that can be done wrong.

Reply:
I would prefer to to take each step separately (for beginners) but I can add a note at the end with the short version.

Reply:
Sure, but my point was that the steps that you copied from Luis are 7 in total, whereas the short version are only three. The end result is more or less the same but with only three steps there is less that can go wrong.

They can surely be written out as individual steps, but I'd rather include the short and simple version.